### DISCUSSION TITLE
How to make NvChad work with cmake projects?

### DISCUSSION THREAD

**Discussion by mwawrzkow** [NvChad/NvChad#2478] [Q&A]
Hi, 

I've installed NvChad out of curiosity, but after some time I've noticed, that in my bit bigger projects where I've got CMakeLists, with CPM modules i get  ie. file not found. 

I'm trying to find the solution on my own, but I think I've started running in circles.

here's an image: 
![image](https://github.com/NvChad/NvChad/assets/28742518/bbd5a16b-bd81-45c7-a0db-e9a680ae82cf)

my project builds, but looks like treesitter is not working as supposed? 
maybe, there's some configuration issue that I'm not aware? 

on the sad side I can see that even vue files with Typescript wont load correctly types:
![image](https://github.com/NvChad/NvChad/assets/28742518/470c6756-d2bb-4b04-81a7-5d7ddd5e2872)

could you help? I will provide all required information. 

Thanks,

**Comment by siduck** [comment]
if treesitter stuff isnt working then do make sure you have tsparsers installed, example, `typescript, tsx, javascript cmake` etc

**Reply by mwawrzkow** [reply]
I suppose I've got them installed, by looking at: 
`TSConfigInfo`
![image](https://github.com/NvChad/NvChad/assets/28742518/dab1be93-332c-4a81-864f-1e1c0a63e6b9)
and `TSInstalInfo`
![image](https://github.com/NvChad/NvChad/assets/28742518/b3c2dda1-72ce-4de8-b2ef-da41986fb44e)

**Reply by siduck** [reply]
btw where's the example of cmake file's syntax highlighting not working? i dont see it

**Reply by mwawrzkow** [reply]
The syntax highlighting works, but the cmake linking and include directories doesnt work:
![image](https://github.com/NvChad/NvChad/assets/28742518/bb439fa2-e1f0-40d3-8fa2-fac3ff7ec7f3)
the CMakeFiles.txt for project:
![image](https://github.com/NvChad/NvChad/assets/28742518/3ea2980d-d218-4080-b488-0b33f52b9b82)
(I'm linking Logger, DatabaseManager  and HTTP libs with their include dirs)
then in code I've should be able to do: 
![image](https://github.com/NvChad/NvChad/assets/28742518/d0625bf7-9247-497b-87a5-c3fcce5431e8)
but I get `Use of undeclared identifier 'LOG'`
but the code compiles as it should be: 
command `cd build && cmake .. && make -j 16 -l 16`
![image](https://github.com/NvChad/NvChad/assets/28742518/8a59a49d-872b-43b3-abbf-4bad89a3aae1)

and then autocompletion and in editor error analysis doesn't work:
![image](https://github.com/NvChad/NvChad/assets/28742518/fed16d74-cfb1-441c-b454-5433c9606d48)
![image](https://github.com/NvChad/NvChad/assets/28742518/a45f1eaa-29e9-43fc-8160-76e8e9b2c281)
The Logger has implemented methods:
![image](https://github.com/NvChad/NvChad/assets/28742518/0790ed40-c3fd-460e-b849-c802516dbcff)
but ie. it doesn't find C++17 types, and can't find CPM linked modules: 
![image](https://github.com/NvChad/NvChad/assets/28742518/2ccba0f8-fc3f-4e95-8348-46cd1ae8dbb0)

**Reply by siduck** [reply]
@mwawrzkow ive never worked with cmake before so cant help, try asking in our discord server or r/neovim.

Anyways how do you manage to even get it working with non nvchad config i.e general neovim config? apply the same logic here

**Reply by mwawrzkow** [reply]
I remember some years ago that I was able to configure that, but yesterday I was looking for out of the box solution. Of Course, I will ask on discord if I don't find any directions on how to do it, I will try to configure it myself from the scratch. Thank you for your time :)
